astropy/modeling/astro_sherpa.py

```diff
@@ -0,0 +1,409 @@
+                    _data.append(Data1DInt("wrapped_data", xlo=xx - xxerr, xhi=xx + xxerr, y=yy))
+            else:
+                dstack = np.dstack((x, y, xerr, yerr))
+                # if dstack.shape[0] == 1:
+                #    dstack = dstack[0]
+                for xx, yy, xxerr, yyerr in _iterTranspose(dstack):
+                    assert xx.shape == yy.shape == xxerr.shape == yyerr.shape, "shape of x, y, xerr and yerr don't match"
+                    _data.append(Data1DInt("wrapped_data", xlo=xx - xxerr, xhi=xx + xxerr, y=yy, staterror=yyerr))
+
+    else:
+        if xerr is None and yerr is None:
+            if zerr is None:
+                dstack = np.dstack((x, y, z))
+                # if dstack.shape[0] == 1:
+                #    dstack = dstack[0]
+                for xx, yy, zz in _iterTranspose(dstack):
+                    assert xx.shape == yy.shape == zz.shape, "shape of x, y, and z don't match"
+                    _data.append(Data2D("wrapped_data", x0=xx, x1=yy, y=zz))
+            else:
+                dstack = np.dstack((x, y, z, zerr))
+                # if dstack.shape[0] == 1:
+                #   dstack = dstack[0]
+                for xx, yy, zz, zzerr in _iterTranspose(dstack):
+                    assert xx.shape == yy.shape == zz.shape == zzerr.shape, "shape of x, y, z and zerr don't match"
+                    _data.append(Data2D("wrapped_data", x0=xx, x1=yy, y=zz, staterror=zzerr))
+        elif xerr is not None and yerr is not None:
+            if zerr is None:
+                dstack = np.dstack((x, y, z, xerr, yerr))
+                # if dstack.shape[0] == 1:
+                #    dstack = dstack[0]
+                for xx, yy, zz, xxerr, yyerr in _iterTranspose(dstack):
+                    assert xx.shape == yy.shape == xxerr.shape == yyerr.shape, "shape of x, y, z, xerr and yerr don't match"
+                    _data.append(Data2DInt("wrapped_data", x0lo=x - xerr, x0hi=x + xerr, x1lo=y - yerr, x1hi=y + yerr))
+            else:
+                dstack = np.dstack((x, y, z, xerr, yerr, zerr))
+                # if dstack.shape[0] == 1:
+                #    dstack = dstack[0]
+                for xx, yy, zz, xxerr, yyerr, zzerr in _iterTranspose(dstack):
+                    assert xx.shape == yy.shape == zz.shape == zzerr.shape == xxerr.shape == yyerr.shape, "shape of x, y, z, xerr, yerr and zerr don't match"
+                    _data.append(Data2DInt("wrapped_data", x0lo=x - xerr, x0hi=x + xerr, x1lo=y - yerr, x1hi=y + yerr, staterror=zerr))
+        else:
+            raise Exception("Set xerr and yerr, or set neither!")
+
+    _ndata = len(_data)
+    if _ndata > 1:
+        _data = DataSimulFit("wrapped_data", _data)
+    else:
+        _data = _data[0]
+    return _data
+
+
+class SherpaFitter(Fitter):
+
+    """
+    Sherpa Fitter for astropy models. Yay :)
+
+    Parameters
+        ----------
+        optimizer : (callable)
+                    A sherpa optimizer method.
```

Comment:
At some point, we probably want to load sherpa docs through intersphinx, and then "`~sherpa.estmethods.EstMethod`" will work. Not sure we want to put that in already (and take sphinx errors until then) or work over again later.